src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -11,33 +10,43 @@
 import teammates.common.datatransfer.FeedbackSessionAttributes;
 import teammates.common.exception.EntityDoesNotExistException;
 import teammates.common.util.ActivityLogEntry;
-import teammates.common.util.Config;
+import teammates.common.util.AdminLogQuery;
 import teammates.common.util.Const;
+import teammates.common.util.GaeVersionApi;
+import teammates.common.util.GaeLogApi;
 import teammates.common.util.StatusMessage;
 import teammates.common.util.TimeHelper;
 import teammates.common.util.Const.StatusMessageColor;
+import teammates.common.util.Version;
 import teammates.logic.api.GateKeeper;
 import teammates.logic.api.Logic;
 
 import com.google.appengine.api.log.AppLogLine;
-import com.google.appengine.api.log.LogQuery;
-import com.google.appengine.api.log.LogServiceFactory;
-import com.google.appengine.api.log.RequestLogs;
-import com.google.appengine.api.log.LogService.LogLevel;
 
 public class AdminActivityLogPageAction extends Action {
-    
-    //We want to pull out the application logs
-    private boolean includeAppLogs = true;
     private static final int RELEVANT_LOGS_PER_PAGE = 50;
-    private static final int SEARCH_TIME_INCREMENT = 2*60*60*1000;  // two hours in millisecond
-    private static final int MAX_SEARCH_TIMES = 12;                 // maximum 1 day
+    /**
+     * The maximum time period to retrieve logs with time increment.
+     */
+    private static final int MAX_SEARCH_PERIOD = 24 * 60 * 60 * 1000; // 24 hrs in milliseconds
+    private static final int SEARCH_TIME_INCREMENT = 2 * 60 * 60 * 1000;  // two hours in milliseconds
+    /**
+     * The maximum number of times to retrieve logs with time increment.
+     */
+    private static final int MAX_SEARCH_TIMES = MAX_SEARCH_PERIOD / SEARCH_TIME_INCREMENT;
+    /**
+     * Maximum number of versions to query.
+     * The current value will include the current version and its 5 preceding versions.
+     */
+    private static final int MAX_VERSIONS_TO_QUERY = 6;
```

Comment:
Consider representing the value as `1 + 5`; redundancy aside, it's in line with the explanation.